### ISSUE TITLE
Self and related links reversed?

### ISSUE THREAD

**Issue by krzkrzkrz** [endpoints/example#20]
The relationship for `http://localhost:8080/v1/authors` looks like:

```
"relationships": {
  "books": {
    "links": {
      "self": "/v1/authors/1/books",
      "related": "/v1/authors/1/relationships/books"
    }
  },
},
```

The relationships for `http://localhost:8080/v1/authors/1` looks like:

```
relationships": {
  "books": {
    "links": {
      "self": "/v1/authors/1/books",
      "related": "/v1/authors/1/relationships/books"
    }
  }
},
```

I might be mistaken. Shouldn't the values of `self` and `related` be the other way around. i.e.

```
"self": "/v1/authors/1/relationships/books",
"related": "/v1/authors/1/books"
```

http://jsonapi.org/recommendations/#urls-relationships says:

> the "relationship URL" - a URL for the relationship itself, which is identified with the self key in a relationship's links object. This URL allows the client to directly manipulate the relationship. For example, it would allow a client to remove an author from a post without deleting the people resource itself.
> 
> the "related resource URL" - a URL for the related resource(s), which is identified with the related key within a relationship's links object. When fetched, it returns the related resource object(s) as the response's primary data.

**Comment by tkellen** [comment]
You're absolutely right. I can't believe this hasn't come up already! I will fix this tonight. @bmac how has this not affected our work?

**Comment by krzkrzkrz** [comment]
I'm curious though, does this affect the current version of Ember Data?

**Comment by bmac** [comment]
~~I do not believe this impacts the current version of Ember Data because Ember Data does not look at this field when generating a url.~~

Edit: See below. async relationships in Ember Data are impacted by this bug.

**Comment by krzkrzkrz** [comment]
@bmac what fields do Ember Data look at to generate the URL?

Second, to clarify, Ember Data does not do anything with the relationship links?

**Comment by bmac** [comment]
Oops I take that back (this is what I get for quickly skimming the issue). 

Ember Data does look at the `related` link when generating the url for "async" relationships. Ember Data never looks at the "self" link and it doesn't use the "related" link when the relationship is not marked with `{ async: true }`. 

@tkellen I believe we haven't hit this before because we are only using sync relationships that are sideloaded using `includes`.

**Comment by krzkrzkrz** [comment]
@bmac, I thought it did too :+1: 

So, without looking at `self`, Ember Data (at this point) cant manipulate the relationship of the parent resource. Am I correct at assuming this?

This might be slightly off-topic now. Was wondering if there was some official/casual doc that outlines how/what Ember Data looks at in the JSON API format. One thing we can put there is the `self` link for model associations when the model has `{ async: true }`, as you described.

Maybe one day, Ember Data will use `self`. That can be included too. Which will probably be triggered via `push`, `set` or something else.

There could be other caveats included, which might help developers use as a reference.

**Comment by tkellen** [comment]
Fixed in 0.9.2. Thanks for bringing this up @krzkrzkrz!
